### PR TITLE
Recheck if service should be deregistered 

### PR DESCRIPTION
**Pull request by janisz**
When service is marked to be deregisterd during sync ask marathon for
tasks to recheck if this task was not started after sync procedure started.
Prevents deregistartion of tasks started after sync asked for all
marathon services. It also minimize call to marathon for a price
of not deregistering tasks that was killed during sync. This will
be handled by events or next sync.

### CODE REVIEW

File: sync/sync_test.go [hunk 1]
```diff
@@ -395,6 +395,29 @@ func TestSync_WithDeregisteringProblems(t *testing.T) {
 	assert.Len(t, services, 1)
 }
 
+func TestSync_WithDeregisteringFallback(t *testing.T) {
```
Comment (adamdubiel) [inline sync/sync_test.go]: we should check what happens if marathon returns error

Comment (janisz) [inline sync/sync_test.go]: Fixed

File: sync/sync.go [hunk 2]
```diff
@@ -211,3 +224,12 @@ func marathonTaskIdsSet(marathonApps []*apps.App) map[apps.TaskID]struct{} {
 	}
 	return tasksSet
 }
+
+func findTask(id apps.TaskID, tasks []*apps.Task) bool {
```
Comment (wendigo) [inline sync/sync.go]: maybe implement this in apps.

```
fun (tasks []*Task) FindById(id TaskID)
```

Comment (janisz) [inline sync/sync.go]: :+1: we could then fix it here [web/event_handler.go:176]https://github.com/allegro/marathon-consul/blob/c96bbf9b4fd5c33bc73b4b097844a43edc256cb4/web/event_handler.go#L176

Comment (janisz) [inline sync/sync.go]: This change will require to change `Marathoner.Tasks` answers from `[]*Task` to `[]Task`. I think it's good change since Task is small object (112 bytes) mostly pointers (4 strings and 2 arrays) so there should not be any problems. This change could be quite big and in multiple packages so I'd like to make it in separate PR #192
